src/java/com/puppetlabs/http/client/impl/JavaClient.java

```diff
@@ -268,7 +278,7 @@ private static void executeWithConsumer(final CloseableHttpAsyncClient client,
                 new StreamingAsyncResponseConsumer(new Deliverable<HttpResponse>() {
             @Override
             public void deliver(HttpResponse httpResponse) {
-                futureCallback.completed(httpResponse);
+                futureCallback.completed(httpResponse); // this stops the timer for the request metric
```

Comment:
FWIW I got confused here because this call also stops a timer context, but it's the one that was in scope when the ```TimedFutureCallback``` was created not the ```timerContext``` in scope here which is a different one.